### DISCUSSION TITLE
Lesson 1: Welcome to Remix! Simple Storage

### DISCUSSION THREAD

**Discussion by vk00007** [smartcontractkit/full-blockchain-solidity-course-py#1659] [Q&A] [answered]
<img width="473" alt="Screenshot 2022-08-21 at 8 33 45 AM" src="https://user-images.githubusercontent.com/111670402/185773514-08242f51-28df-40a0-a215-958585ef793d.png">

In the FCC blockchain tutorial, why can't we have uint256 returning in the function retrieve, why is it separately returns(uint256).  Is returns another function here? if so why is retrieve() used here at all.

PS: An absolute beginner here, forgive me if this is very basic

**Comment by cromewar** [comment, marked as answer]
Hello @vk00007 
In solidity the `returns` statement is just to specify the type of data you are going to return, the reason we create this function is because you will not always use REMIX, as favoriteNumber is a public variable, Remix automatically set up a button for you, but when you are going to develop locally you will not have it, so the retrieve function is a `getter` function.

**Reply by Odinaka12** [reply]
Pls help me on this am a beginner just started yesterday.
This array function keep showing error in remix 
Help me check if I am doing it the wrong way 🙏
![IMG_20220826_143948](https://user-images.githubusercontent.com/99669756/186921151-28ee96d1-e36d-44c8-b2be-51fe3b204689.jpg)

**Reply by Niferu** [reply]
Hello @Odinaka12 

Please use code editor to pass your code next time instead of photo of screen. Your curly brackets "{ }" are posted incorrectly. You have closed your contract coding just in line 8, and you should closed it after all functions and variables.

**Comment by vk00007** [comment]
Thank you, I get it now.

**Comment by solidityb** [comment]
Hi,

Appreciate if someone can help me here. I tried to search for JavaScript, but couldn't. What did I do wrongly?

![image](https://user-images.githubusercontent.com/114201152/191878575-82f10d24-f0e1-4819-838f-b1e65991e17c.png)

Thanks.

**Reply by cromewar** [reply]
Hello @solidityb 
For Javascript? could you be more specific on the question?
If you mean to use the integrated blockchain on remix, using (London or Berlin) should be enough.

**Reply by Niferu** [reply]
Hello @cromewar 
I bet he means that Patrick on video has below:
![image](https://user-images.githubusercontent.com/110537615/192025690-23903a13-2324-42a2-bf06-7ab64dd6af98.png)
 I'm not sure if it is his custom VM or not but I have already replied in separate dicussion posted by @solidityb that using Remix VM (London) will be the same.

**Reply by solidityb** [reply]
Yes, I was following the video, got confused. Thanks :)
